wlm/src/main/java/ai/djl/serving/wlm/ModelInfo.java

```diff
@@ -1000,6 +1012,92 @@ private void downloadS3(String src, String dest) throws ModelException {
         }
     }
 
+    private void initTrtLlmModel() throws ModelException, IOException {
+        // check if downloadS3Dir or local model path is a trt-llm repo
+        boolean isTrtLlmRepo = isValidTrtLlmModelRepo();
+        if (!isTrtLlmRepo) {
+            build_trt_llm_engine();
+        }
+    }
+
+    private void build_trt_llm_engine() throws ModelException, IOException {
+        logger.info("Converting model to TensorRT-LLM artifacts");
+        trtLlmRepoDir = Paths.get("/tmp/tensorrtllm");
+        // invoke trt-llm build script
+        List<String> commandList = new ArrayList<>();
+        commandList.add("python");
+        commandList.add("/opt/djl/partition/trt_llm_partition.py");
+        commandList.add("--properties_dir");
+        commandList.add(modelDir.toAbsolutePath().toString());
+        commandList.add("--trt_llm_model_repo");
+        commandList.add(trtLlmRepoDir.toAbsolutePath().toString());
+        if (downloadS3Dir != null) {
+            commandList.add("--model_path");
+            commandList.add(downloadS3Dir.toAbsolutePath().toString());
+        }
+        try {
+            Process exec = new ProcessBuilder(commandList).redirectErrorStream(true).start();
+                String logOutput;
+                try (InputStream is = exec.getInputStream()) {
+                    logOutput = Utils.toString(is);
+                }
+                int exitCode = exec.waitFor();
+                if (0 != exitCode || logOutput.startsWith("ERROR ")) {
+                    logger.error(logOutput);
+                    throw new EngineException("Download model failed.");
+                } else {
+                    logger.info(logOutput);
+                }
+            } catch (IOException | InterruptedException e) {
+            throw new ModelException("Failed to build TensorRT-LLM engine", e);
+        }
+        logger.info("TensorRT-LLM engine built successfully");
+    }
+
+    private boolean isValidTrtLlmModelRepo() throws IOException {
+        Path dirToCheck = null;
+        Path modelPath = Paths.get(prop.getProperty("option.model_id"));
+        if (downloadS3Dir != null) {
+            dirToCheck = downloadS3Dir;
+        }
+        logger.info("modelPath: {}", modelPath);
+        logger.info("exists: {}", Files.exists(modelPath));
+        if (Files.exists(modelPath)) {
+            dirToCheck = modelPath;
+        }
+        if (dirToCheck == null) {
+            return false;
+        }
+
+        List<Path> configFiles = new ArrayList<>();
+        List<Path> engineFiles = new ArrayList<>();
+        List<Path> tokenizerFiles = new ArrayList<>();
+        try (Stream<Path> walk = Files.walk(dirToCheck)) {
+            walk.filter(Files::isRegularFile).forEach(
+            path -> {
+                if (path.getFileName().toString().endsWith(".engine")) {
+                    engineFiles.add(path);
```

Comment:
I think we can just check pbtxt files. No need to walk with engine files in case some customers have special structured folders

Reply:
this doesn't check for folder structures, it checks recursively in sub directories and looks for files ending with `.engine`. I am okay to just check `config.pbtxt` as it's hard to enforce full repo structure and customers should take some of that responsibility.

Reply:
Updated to check config.pbtxt and it's presence at depth=2